packages/manager/apps/pci-kubernetes/src/api/hooks/kubernetes.ts

```diff
@@ -0,0 +1,47 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+  getKubernetesCluster,
+  updateKubernetesCluster,
+} from '@/api/data/kubernetes';
+import queryClient from '@/queryClient';
+
+export const useKubernetesCluster = (projectId: string, kubeId: string) =>
+  useQuery({
+    queryKey: ['project', projectId, 'cluster', kubeId],
+    queryFn: () => getKubernetesCluster(projectId, kubeId),
+  });
+
+type RenameKubernetesClusterProps = {
+  projectId: string;
+  kubeId: string;
+  name: string;
+  onError: (cause: Error) => void;
+  onSuccess: () => void;
+};
+export const useRenameKubernetesCluster = ({
+  projectId,
+  kubeId,
+  name,
+  onError,
+  onSuccess,
+}: RenameKubernetesClusterProps) => {
+  const mutation = useMutation({
+    mutationFn: async () =>
+      updateKubernetesCluster(projectId, kubeId, { name }),
+    onError,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
+        queryKey: ['project', projectId, 'cluster', kubeId],
```

Comment:
You can refactor the key into a separate function to reuse it whenever you want.